### ISSUE TITLE
Creating namespace  "kube-system" does not result in "kube-system" environment being created.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#5863]
Rancher Version: v1.2.0-pre2

Setup - k8s envitronment with 3 hosts.
K8s version - v1.2.4

Steps to reproduce the problem:
```
kubectl create namespace kube-system
kubectl create -f test.yml --namespace kube-system
```
On this first attempt , a new stack - "kube-system" was created on the rancher-server side.
Ingress was created with targets as expected.

kubectl delete namespace kube-system.
This does not result in "kube-system" stack being deleted on the rancher-server side.

Manually removed the environment - "kube-system" stack.

Create namespace kube-system again.
```
kubectl create namespace kube-system
kubectl create -f test.yml --namespace kube-system
```
After this , I dont see the a new stack "kube-system" being created on the rancher-server side.

ingress is not able to add targets from "kube-system" namespace

<img width="1263" alt="screen shot 2016-08-31 at 4 04 01 pm" src="https://cloud.githubusercontent.com/assets/4266958/18149362/945f0f16-6f94-11e6-9b22-652bd364df6e.png">

Following error is seen in repeatedly in Kubernetes_rancher-ingress-controller logs:
```
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:10 PMtime="2016-08-31T23:05:10Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:11 PMtime="2016-08-31T23:05:11Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:12 PMtime="2016-08-31T23:05:12Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:12 PMtime="2016-08-31T23:05:12Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
8/31/2016 4:05:12 PMtime="2016-08-31T23:05:12Z" level=error msg="Failed to apply lb config on provider: Coudln't get stack by name [kube-system]"
```

**Comment by sangeethah** [comment]
"kube-system" environment does not get recreated in the following scenarios as well:

Scenario 1:

1. Create namespace kube-system and deploy few services
```
kubectl create namespace kube-system
kubectl create -f test.yml --namespace kube-system
```
Note that "kube-system" environment is created.

2. Make this k8s environment a  cattle environment.
Note that "kube-system" environment gets marked as "removed".

3. Make this k8s environment a  k8 environment and try to create and namespace kube-system and deploy few services in it.

You will notice that the Ingress rule gets created with no targets. "kube-system" environment does not get created in this case.

Scenario 2:

1. Create namespace kube-system and deploy few services
```
kubectl create namespace kube-system
kubectl create -f test.yml --namespace kube-system
```
Note that "kube-system" environment is created.

2. Make this k8s environment a  cattle environment.
Note that "kube-system" environment gets marked as "removed".

3. Create a new k8s environment .
create namespace kube-system and deploy few services in it.

You will notice that the Ingress rule gets created with no targets. "kube-system" environment does not get created in this case.

**Comment by wlan0** [comment]
@sangeethah This issue is fixed by a cattle fix (https://github.com/rancher/cattle/commit/8ef32384fcfa592e011802cb49e785078e1ea8fb ) and a kubernetes agent fix (https://github.com/rancher/kubernetes-agent/commit/f7b0761e728b87c1286c959c341d300169e43147)

The cattle fix is in v1.2.0-pre3-rc1
I've added the agent fix to this qa template - https://github.com/rancher/rancher-catalog/commit/356b24f928fca5d2425adf6c09533073d5dcd5e0 (Folder number 12)

**Comment by sangeethah** [comment]
@soumyalj, when you verify this issue , include the K8s version ( and template version)  that was used to validate this issue since the fixes may not be available for  k8s 1.3 version.

**Comment by soumyalj** [comment]
Tested on master with Kubernetes 1.4 with template version 1.4-rancher1.
Followed the steps above. I was able to create namespace, services/ingress in it, delete the namespace and recreate a namespace with the same name. Ingresses were functional too.
mysql> select id,name,state,created,removed,account_id from environment;
+----+------------------------+---------+---------------------+---------------------+------------+
| id | name                   | state   | created             | removed             | account_id |
+----+------------------------+---------+---------------------+---------------------+------------+
|  1 | Kubernetes             | active  | 2016-09-30 01:52:08 | NULL                |          5 |
|  2 | default                | active  | 2016-09-30 01:56:07 | NULL                |          5 |
|  3 | kubernetes-ingress-lbs | active  | 2016-09-30 01:56:23 | NULL                |          5 |
|  4 | hello123               | removed | 2016-09-30 02:29:48 | 2016-09-30 02:31:00 |          5 |
|  5 | hello123               | removed | 2016-09-30 23:15:08 | 2016-09-30 23:26:55 |          5 |
|  6 | hello123               | active  | 2016-09-30 23:28:03 | NULL                |          5 |
+----+------------------------+---------+---------------------+---------------------+------------+
6 rows in set (0.00 sec)
